content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,122 @@
+
+`InTreePlugin{vendor}Unregister` is a standalone feature gate that can be enabled and disabled out of CSI Migration scope. The name speaks for itself, when enabled, the component will not register the specific in-tree storage plugin to the supported list. If the cluster operator only enables this flag, end users will get an error from PVC saying it cannot find the plugin when the plugin is used. The cluster operator may want to enable this regardless of CSI Migration if they do not want to support the legacy in-tree APIs and only support CSI moving forward.
+
+At the same time, `CSIMigration{provider}Complete` was deprecated and removed in kubernetes v1.21 as well. The `InTreePlugin{vendor}Unregister` will replace the old feature flag but remain all the functionality that `CSIMigration{provider}Complete` provides.
+
+### Observability
+
+We have added the [CSI Migration Metrics](https://github.com/kubernetes/kubernetes/issues/98279) support for better observability of the feature. This also help us discover any potential issues regarding the feature.
+
+- storage_operation_duration_seconds
+  - This metric is exposed from the Kube-Controller-Manager. A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_sidecar_operations_seconds
+  - This metric is exposed from the CSI sidecars(csi-provisioner, csi-attacher, csi-resizer). A new field `migrated` is added to the metric to indicate whether this storage operation is a CSI migration operation.
+- csi_kubelet_operations_seconds
+  - This metric is exposed from Kubelet. It exposes CSI operation, driver  name, status and `migrated` field to indicate whether this storage operation is a CSI migration operation.
+
+### Bug fixes and feature improvement
+
+We have been able to identify bugs like dangling attachment, garbage collecting, incorrect topology label ... and we have fixed all of them to make the feature more health.
+
+### Cloud Provider && Cluster Lifecycle Collaboration
+
+We, Sig-Storage, has been working closely with Sig-CloudProvider and Sig-ClusterLifeCycle to provide a better experience for the feature launching.
+
+If you are a user from public cloud provider managed Kubernetes service like GKE, EKS, AKS, there is nothing need to be done from your end. The cloud providers will manage the feature gate and ensure the feature lands correctly.
+
+If you are a user of some other kubernetes distribution services, check the official documentations to get support for the feature. We have communicate with the Sig-ClusterLifeCycle to request early planing for CSI Migration GA and they have been working hard to ensure a smooth experience.
+
+## What is the timeline/status?
+
+The current and targeted releases for each individual cloud driver is shown in the table below:
```

Comment:
Some of the drivers listed below are not for cloud providers.  So "cloud driver" -> "driver"